### PR TITLE
Set stock quantity value as 0 by default

### LINKED ISSUE #46342: [Enhancement]: Default the stock quantity to 0
Revert the changes added in https://github.com/woocommerce/woocommerce/issues/37117 and default the stock quantity to 0.

When a product with enabled stock management is added without specifying a stock value, the product form defaults the stock quantity to 1, whereas in other areas of the system it's null. This inconsistency leads to unexpected increases in stock quantity to 1 when the product is subsequently edited and saved by a user, without any action taken to change the stock value.

The expected behavior is to automatically assign a zero value to the stock instead of `null` or `1`.

For additional details https://github.com/woocommerce/woocommerce/issues/39475

### PR DESCRIPTION
**Pull request by octaedro**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR reverts the changes added in https://github.com/woocommerce/woocommerce/issues/37117 and defaults the stock quantity to 0.

We used to default the stock quantity to 1 when a product with enabled stock management was added without specifying a stock value. However, it was `null` in other areas of the system. This inconsistency was problematic and led to unexpected behaviors.

The changes added here:
- Set the stock quantity to 0 by default.
- It's not possible [to set the stock quantity to `null`](https://github.com/woocommerce/woocommerce/pull/48448/files#diff-d62350bd045581ade84f14b7a7600040162cc905417e817a9cae27cfd3ac93aeR969) anymore
- `get_stock_quantity` doesn't return `null` anymore. It will return an integer. [#](https://github.com/woocommerce/woocommerce/pull/48448/files#diff-ef27d662bcec590c665fc7ac8c4a698c5a0a40af67376751fa2299c2541c0a79R341) and [#](https://github.com/woocommerce/woocommerce/pull/48448/files#diff-d62350bd045581ade84f14b7a7600040162cc905417e817a9cae27cfd3ac93aeR364)

For additional details https://github.com/woocommerce/woocommerce/issues/39475

Screenshot 2024-06-14 at 8 54 57 AM

and

Screenshot 2024-06-14 at 8 55 51 AM

Closes #46342.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch.
2. Go to Products > Add New > Inventory.
3. Check `Track stock quantity for this product`.
4. Verify that `Quantity` is 0.
5. Create a product and confirm that the stock management works well.
6. Try also a few extensions that interacts with the stock quantity like `WooCommerce Product Bundles` and `WooCommerce Back in Stock Notifications`.
7.  Make some smoke tests.
8. Make sure that `Try the new product editor (Beta)` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
9. Go to Products > Add New > Inventory.
10. Press `Track inventory`.
11. Verify the `Available stock` is 0.
12. Make some smoke tests.
13. Test with variable products as well.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-product-data.php [hunk 1]
```diff
@@ -197,7 +197,7 @@ private static function prepare_downloads( $file_names, $file_urls, $file_hashes
 		if ( ! empty( $file_urls ) ) {
 			$file_url_size = count( $file_urls );
 
-			for ( $i = 0; $i < $file_url_size; $i ++ ) {
+			for ( $i = 0; $i < $file_url_size; $i++ ) {
```
Comment (octaedro) [inline plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-product-data.php:200]: This line has been modified automatically by the linter.

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-inventory.php [hunk 2]
```diff
@@ -71,10 +71,10 @@
...
 
 			$backorder_args = array(
-				'id'          => '_backorders',
-				'value'       => $product_object->get_backorders( 'edit' ),
-				'label'       => __( 'Allow backorders?', 'woocommerce' ),
-				'options'     => wc_get_product_backorder_options(),
+				'id'      => '_backorders',
+				'value'   => $product_object->get_backorders( 'edit' ),
+				'label'   => __( 'Allow backorders?', 'woocommerce' ),
+				'options' => wc_get_product_backorder_options(),
```
Comment (octaedro) [inline plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-inventory.php:77]: [Same](https://github.com/woocommerce/woocommerce/pull/48448/files#r1639764534) with these lines.

File: plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php [hunk 3]
```diff
@@ -1236,7 +1236,7 @@ public function create_all_product_variations( $product, $limit = -1, $default_v
 
 			do_action( 'product_variation_linked', $variation_id );
 
-			$count ++;
+			++$count;
```
Comment (octaedro) [inline plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php:1239]: [Same](https://github.com/woocommerce/woocommerce/pull/48448/files#r1639764534) with this line.

**Review by mdperez86 — APPROVED** [review]
LGTM @octaedro nice job!!!

**Pull request merged**